iwyu_test_util.py

```diff
@@ -92,6 +134,32 @@ def HasTarget(target):
         supported.append(m.group(1))
     return target in supported
 
+  @staticmethod
+  def UsingCXXstdlib(lib):
+    id_macro = {
+      'libcxx' : '_LIBCPP_VERSION',
+      'libstdcxx' : '__GLIBCXX__',
+      'msvc' : '_MSVC_STL_UPDATE', # https://github.com/microsoft/STL/wiki/Macro-_MSVC_STL_UPDATE
+    }
+    if lib not in id_macro:
+      assert False, ('Unknown STL: %s' % lib)
```

Comment:
I don't think we should assert on user input.  How about `raise SyntaxError(...)` instead, even if we don't have file/line information?

Same for `UsingCstdlib` below.